yt_dlp/extractor/kukululive.py

```diff
@@ -0,0 +1,125 @@
+                'ext': 'mp4',
+                'vcodec': vcodec,
+                'quality': quality_priority[quality] - 5,
+            },
+            {
+                'format_id': f'{quality}-audioonly',
+                'url': quality_meta.get('hlsaddr_audioonly')[0],
+                'ext': 'aac',
+                'vcodec': 'none',
+                'quality': quality_priority[quality] - 10,
+            },
+        ])
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        html = self._download_webpage(url, video_id)
+
+        title = clean_html(get_element_by_id('livetitle', html.replace('<SPAN', '<span').replace('SPAN>', 'span>')))
+        description = self._html_search_meta('Description', html)
+        thumbnail = self._html_search_meta(['og:image', 'twitter:image'], html)
+
+        is_live_stream = 'var timeshift = false;' in html
+        is_vod = 'var timeshift = true;' in html
+
+        if is_live_stream:
+            qualities = [
+                ('high', 'Z'),
+                ('low', 'ForceLow'),
+            ]
+            formats = []
+            for (desc, code) in qualities:
+                quality_meta = self._get_quality_meta(video_id, desc, code)
+                self._add_quality_formats(formats, quality_meta)
+                if desc == 'high' and quality_meta.get('vcodec')[0] == 'HEVC':
+                    h264_meta = self._get_quality_meta(video_id, desc, code, force_h264='1')
+                    self._add_quality_formats(formats, h264_meta)
+
+            return {
+                'id': video_id,
+                'title': title,
+                'description': description,
+                'thumbnail': thumbnail,
+                'is_live': True,
+                'formats': formats,
+            }
+
+        if is_vod:
+            player_html = self._download_webpage(
+                'https://live.erinn.biz/live.timeshift.fplayer.php', video_id,
+                query={'hash': video_id},
+                note='Downloading player html',
+                errnote='Unable to download player html')
+
+            # https://regex101.com/r/3AXpSA/3
+            sources = self._search_regex(r'var fplayer_source = ([^;]+)', player_html, 'sources')
+            sources_json = self._parse_json(sources.replace('.mp4",', '.mp4"'), video_id)
+
+            formats = []
+            for source in sources_json:
+                path = source.get('file')
+                formats.append({
+                    'url': urljoin('https://live.erinn.biz', path),
+                    'ext': 'mp4',
+                    'protocol': 'm3u8_native',
+                })
+
+            return {
+                'id': video_id,
+                'title': title,
+                'description': description,
+                'timestamp': sources_json[0].get('time_start'),
+                'thumbnail': thumbnail,
+                'formats': formats,
+            }
```

Comment:
I think these are not separate formats of the same video, but segments of the stream, say if the connection dropped for part of the stream.
So maybe returning this as a multi video would make sense:
```suggestion
            entries = []
            for i, segment in enumerate(sources_json):
                path = segment.get('file')
                if not path:
                    continue
                formats = [{
                    'url': urljoin('https://live.erinn.biz', path),
                    'ext': 'mp4',
                    'protocol': 'm3u8_native',
                }]
                entries.append({
                    'id': f'{video_id}_{i}',
                    'title': f'{title} (Part {i + 1})',
                    'description': description,
                    'timestamp': traverse_obj(segment, ('time_start', {int_or_none})),
                    'thumbnail': thumbnail,
                    'formats': formats,
                })
            return self.playlist_result(entries, video_id, title, description, multi_video=True)
```
Potentially, we would want some additional logic to return it as a single video if there is only one segment.